cosmos/providers/dbt/render.py

```diff
@@ -56,8 +59,24 @@ def render_project(
     # iterate over each model once to create the initial tasks
     for model_name, model in project.models.items():
         # if we have tags, only include models that have at least one of the tags
-        if dbt_tags and not set(dbt_tags).intersection(model.config.tags):
-            continue
+
+        # ensures the same tag isn't in select & exclude
+        if "tags" in select and "tags" in exclude:
+            if set(select["tags"]).intersection(exclude["tags"]):
+                raise AirflowException(
+                    f"Can't specify the same tag in `select` and `include`: "
+                    f"{set(select['tags']).intersection(exclude['tags'])}"
+                )
```

Comment:
we should be able to move this up, outside of the for loop for efficiency (the input tags won't change across each model)

Reply:
Done.